src/dom.js

```diff
@@ -1101,27 +1101,53 @@ export function getShadowHost(element : Node) : ?HTMLElement {
     }
 }
 
-export function insertShadowSlot(element : HTMLElement) : HTMLElement {
+  
+export function getRootNode(element : Node) : Node {
+    while (element.parentNode) {
+        element = element.parentNode;
+    }
+  
+    return element;
+}
+
+export function insertShadowSlot(element : HTMLElement, styles? : HTMLElement) : HTMLElement {
+    const DEFAULT_STYLESHEET_INDEX = 0;
     const shadowHost = getShadowHost(element);
 
     if (!shadowHost) {
         throw new Error(`Element is not in shadow dom`);
     }
 
-    if (isShadowElement(shadowHost)) {
-        throw new Error(`Host element is also in shadow dom`);
-    }
-
-    const slotName = `shadow-slot-${ uniqueID() }`;
+    // $FlowFixMe
```

Comment:
this comment is needed since flow assumes that node objects don´t have `querySelector` method

Reply:
Thanks for trying. Let's leave this one then.